os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -402,12 +393,12 @@ static int imxrt_erase(FAR struct mtd_dev_s *dev, off_t startsector, size_t nsec
 
 		/* Not using sub-sector erase.  Erase each full sector */
 
-		imxrt_sectorerase(priv, startsector + sectors_erased);
+		imxrt_sectorerase(priv, startblock + sectors_erased);
```

Comment:
In imxrt_sectorerase, it calculates offset based on sector. so you should consider it.

Reply:
Okay, I changed "startblock" into "startsector"...